operator/controllers/loki/internal/lokistack/update.go

```diff
@@ -41,3 +41,35 @@ func updateAnnotation(ctx context.Context, k k8s.Client, stack *lokiv1.LokiStack
 		return k.Update(ctx, stack)
 	})
 }
+
+func removeAnnotation(ctx context.Context, k k8s.Client, stack *lokiv1.LokiStack, key string) error {
+	if stack.Annotations == nil {
+		return nil
+	}
+	delete(stack.Annotations, key)
+
+	err := k.Update(ctx, stack)
+	switch {
+	case err == nil:
+		return nil
+	case errors.IsConflict(err):
+		// break into retry logic below on conflict
+		break
+	case err != nil:
+		return err
+	}
+
+	objectKey := client.ObjectKeyFromObject(stack)
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
I don't quite understand on what conditions this code ever passes the switch case and executes `RetryOnConflict`. This seems never to be the case to me.

Reply:
That case here is when there is a conflict (same as for `updateAnnotation`), specifically the case `errors.IsConflict(err)`.

Reply:
That makes sense now that you are telling me... I looked at `break` and read `return` :facepalm: